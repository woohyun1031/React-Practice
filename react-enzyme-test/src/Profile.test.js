import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="woo1031" name="김우현" />);
    expect(wrapper).toMatchSnapshot();
  });
});

//mount : Enzyme을 통하여 리액트 컴포넌트를 렌더링
//이를 통해서 만든 wrapper 를 통해서 우리가 추후 props 조회, DOM 조회, state 조회 등을 할 수 있음
