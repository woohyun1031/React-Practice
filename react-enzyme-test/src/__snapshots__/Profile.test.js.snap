// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="김우현"
  username="woo1031"
>
  <div>
    <b>
      woo1031
    </b>
     
    <span>
      (
      김우현
      )
    </span>
  </div>
</Profile>
`;
